addons/l10n_fr_certification/models/account_closure.py

```diff
@@ -0,0 +1,153 @@
+                AND m.state = 'posted'
+                AND m.write_date '''
+
+        if date_start and date_stop:
+            params['date_stop'] = date_stop
+            params['date_start'] = date_start
+            query += 'BETWEEN %(date_start)s AND %(date_stop)s'
+        elif not date_start:
+            params['date_stop'] = date_stop
+            query += '<= %(date_stop)s'
+
+        if avoid_move_ids:
+            params['avoid_move_ids'] = tuple(avoid_move_ids)
+            query += 'AND m.id NOT IN %(avoid_move_ids)s'
+
+        return query, params
+
+    def _query_for_aml(self, company, date_start='', date_stop='', avoid_move_ids=[]):
+        query, params = self._build_query_for_aml(company, date_start, date_stop, avoid_move_ids)
+        self.env.cr.execute(query, params)
+        return self.env.cr.dictfetchall()
+
+    def _compute_amounts(self, frequency, company):
+        interval_dates = self._interval_dates(frequency, company)
+        previous_closure = self.search([
+            ('frequency', '=', frequency),
+            ('company_id', '=', company.id)], limit=1, order='sequence_number desc')
+
+        date_query_start = ''
+        previous_date = False
+        if previous_closure and previous_closure.move_ids:
+            previous_date = previous_closure.move_ids.sorted(key=lambda m: m.write_date)[-1].write_date
+            date_query_start = min(previous_date, interval_dates['interval_from'])
+
+        moves_already_counted = self._query_closures_for_move_ids(frequency, company)
+        aml_fetched = self._query_for_aml(company, date_query_start, interval_dates['date_stop'], moves_already_counted)
+
+        date_interval_start = date_query_start and date_query_start or interval_dates['interval_from']
+        aml_interval = filter(lambda aml: aml['move_write_date'] >= date_interval_start, aml_fetched)
+
+        total_interval = sum(aml['balance'] for aml in aml_interval)
+        total_beginning = sum(aml['balance'] for aml in aml_fetched)
+
+        return {'total_interval': total_interval,
+                'total_beginning': previous_closure.total_beginning + total_beginning,
+                'move_ids': [(4, aml['move_id'], False) for aml in aml_interval],
+                'date_closure_stop': interval_dates['date_stop'],
+                'date_closure_start': previous_date and previous_date or interval_dates['interval_from']}
+
+    def _interval_dates(self, frequency, company):
+        date_stop = datetime.utcnow()
+        interval_from = None
+        if frequency == 'daily':
+            interval_from = date_stop - timedelta(days=1)
+        elif frequency == 'monthly':
+            month_target = date_stop.month > 1 and date_stop.month - 1 or 12
+            year_target = month_target < 12 and date_stop.year or date_stop.year - 1
+            interval_from = date_stop.replace(year=year_target, month=month_target)
+        elif frequency == 'annually':
```

Comment:
annually frequency should probably use compute_fiscalyear_dates() from res.company instead, as you probably want to  make the closure till the end of the fiscal year. (if called on the 31st of December at 4AM, you'll miss the last day in your current implementation, I suppose)

Reply:
Nope, annually literally means annually. We do want a report on the last ~365 days.
The law gives us the option to take fiscal year OR annually

Reply:
ok but why would you want using the last 365 days instead of the fiscal year?? I don't see any benefit to it, it just makes the closure length inconsistent...

Reply:
it is essentially a choice, but it saves some headache:
- say the previous fiscal year is over
- you make  move today for the previous fiscal year
- that move will go unseen anywhere it the case we choose fiscal year
- hence grand total will always be false